src/main/java/org/cardanofoundation/explorer/api/repository/ledgersyncagg/AddressTxAmountRepository.java

```diff
@@ -198,8 +198,11 @@ List<AddressTxAmount> findAllByStakeAddressAndTxHashIn(
   @Query(
       value =
           """
-              SELECT max(atm.blockTime) FROM AddressTxAmount atm
-              WHERE atm.unit = :unit
+              WITH latest_token_slot AS ( SELECT MAX(slot) AS slot FROM AddressTxAmount WHERE unit = :unit )
+              SELECT max(blockTime)
+              FROM AddressTxAmount
+              WHERE unit = :unit
+              AND slot = (SELECT slot FROM latest_token_slot)
```

Comment:
How about we select the maximum token slot and then use the java CardanoConverter to convert the slot to block_time ?

Reply:
yup, makes perfect sense

Reply:
This should work, right?
`AddressTxAmount findTopByUnitOrderBySlotDesc(String unit);`

Reply:
Let me check in my side and update code back to you :D

Reply:
Oops, you already did :D

Reply:
new code make sence to me